### PR TITLE
Don't automaticallly report resource timing for cross-origin TAO-fail…

### PR DESCRIPTION
**Pull request by noamr**
… iframes

Reporting the response end time for an iframe may leak information about user interaction with that iframe.

See w3c/resource-timing#340

- [x] At least two implementers are interested (and none opposed):
   * Chromium
   * WebKit
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * https://github.com/web-platform-tests/wpt/pull/37810
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1404695
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1809182
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=250322
   * Deno (not for CORS changes): …
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: https://github.com/mdn/content/issues/23740

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)

***
<a href="https://whatpr.org/fetch/1579.html" title="Last updated on Jan 25, 2023, 1:13 PM UTC (c3cff9d)">Preview</a> | <a href="https://whatpr.org/fetch/1579/e1e9567...c3cff9d.html" title="Last updated on Jan 25, 2023, 1:13 PM UTC (c3cff9d)">Diff</a>

### CODE REVIEW

File: fetch.bs [hunk 1]
```diff
@@ -4667,11 +4667,25 @@ steps:
 
      <li>
       <p>If <var>response</var>'s <a for=response>timing allow passed flag</a> is not set,
-      then set <var>timingInfo</var> to the result of <a>creating an opaque timing info</a> for
-      <var>timingInfo</var>, set <var>bodyInfo</var> to a new <a for=/>response body info</a>, and
-      set <var>cacheState</var> to the empty string.
-
-      <p class=note>This covers the case of <var>response</var> being a <a>network error</a>.
+      then:
+       <ol>
```
Comment (annevk) [inline fetch.bs:4671]: Indented too much by 1. (Goes for the entire block.) Could also use a newline before the `<ol>`.

File: fetch.bs [hunk 2]
```diff
@@ -4667,11 +4667,25 @@ steps:
...
+        <li>
+         <p>If <var>fetchParams</var>'s
+         <a for="fetch params">request</a>'s <a for=request>mode</a> is "<code>navigate</code>",
+         then abort these <a for="fetch controller">report timing steps</a>.
```
Comment (annevk) [inline fetch.bs:4675]: I think we should just say "abort these steps".

File: fetch.bs [hunk 3]
```diff
@@ -4667,11 +4667,25 @@ steps:
...
+
+         <p class=note>Reporting timing information for cross-origin navigations without
+         `<code>Timing-Allow-Origin</code>` may expose information about user interaction with that
+         origin.
+
+        <li>
+         <p>Set <var>timingInfo</var> to the result of <a>creating an opaque timing info</a> for
+          <var>timingInfo</var>, set <var>bodyInfo</var> to a new <a for=/>response body info</a>,
```
Comment (annevk) [inline fetch.bs:4683]: Indentation went wrong.

Comment (noamr) [reply fetch.bs]: Fixed

File: fetch.bs (hunk 2 above)
Comment (noamr) [reply fetch.bs]: Fixed

File: fetch.bs (hunk 1 above)
Comment (noamr) [reply fetch.bs]: Fixed

File: fetch.bs [hunk 4]
```diff
@@ -4677,11 +4677,26 @@ steps:
...
 
-      <p class=note>This covers the case of <var>response</var> being a <a>network error</a>.
+      <ol>
+       <li>
+        <p>If <var>fetchParams</var>'s
+        <a for="fetch params">request</a>'s <a for=request>mode</a> is "<code>navigate</code>",
+        then abort these steps.
+
+        <p class=note>Reporting timing information for cross-origin navigations without
+        `<code>Timing-Allow-Origin</code>` may expose information about user interaction with that
```
Comment (annevk) [inline fetch.bs:4689]: without the Timing-Allow-Origin header*

File: fetch.bs [hunk 5]
```diff
@@ -4677,11 +4677,26 @@ steps:
...
+        origin.
```
Comment (annevk) [inline fetch.bs:4690]: Thinking about this again the description here is misleading as it's not just user interaction but general nested document navigations that can be problematic, right?

**Review by annevk — COMMENTED** [review]
Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?

**Comment by noamr** [comment]
> Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?

Right, it's combined with an HTML PR.

File: fetch.bs (hunk 4 above)
Comment (noamr) [reply fetch.bs:4689]: Fixed

**Comment by noamr** [comment]
> Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?

@annevk, For the OP, how do I go about WebKit's position on this? (together with https://github.com/whatwg/html/pull/8643)
This is basically a spec of the idea [in this comment](https://github.com/w3c/resource-timing/issues/340#issuecomment-1211002343).

**Comment by noamr** [comment]
> If this is something WebKit already implements (I don't know) I think it's okay to assume WebKit is supportive of making it more restrictive.

Yes, WebKit already implements TAO-protected iframes.

**Comment by noamr** [comment]
> Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?

OP is completed and nits addressed, for this and for the HTML PR.

**Comment by noamr** [comment]
> > Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?
> 
> OP is completed and nits addressed, for this and for the HTML PR.

^^ @annevk

**Comment by noamr** [comment]
> > Overall this looks good, couple nits on the first note. And I guess OP still has to be completed, right?
> 
> OP is completed and nits addressed, for this and for the HTML PR.

^^ @annevk

**Pull request closed without merging**